dasharo-security/secure-boot.robot

```diff
@@ -129,281 +121,210 @@ SBO003.001 Attempt to boot file with the correct key from Shell (firmware)
     [Documentation]    PEM generated with `openssl req -new -x509 -newkey rsa:3072 -subj "/CN=DB-RSA3072/" -keyout DB-RSA3072.key -out DB-RSA3072.pem -days 3650 -nodes -sha256`
     ...    converted to DER using `openssl base64 -d -in DB-RSA3072.cer -out DB-RSA3072.der`
-    Mount ISO As USB In QEMU    ${CURDIR}/../scripts/secure-boot/images/RSA3072.img
+    Skip If    not ${SECURE_BOOT_SUPPORT}    SBO010.002 not supported
+    Skip If    not ${TESTS_IN_FIRMWARE_SUPPORT}    SBO010.002 not supported
+    Mount ISO As USB    ${CURDIR}/../scripts/secure-boot/images/RSA3072.img
     Power On
 
     ${sb_menu}=    Enter Secure Boot Menu And Return Construction
     Enable Secure Boot    ${sb_menu}
-    ${sb_menu}=    Reenter Menu And Return Construction
-    ${advanced_menu}=    Enter Advanced Secure Boot Keys Management And Return Construction    ${sb_menu}
-    Enter Enroll DB Signature Using File In DB Options    ${advanced_menu}
-    Enter Volume In File Explorer    RSA3072
-    Select File In File Explorer    cert.der
-    Save Changes And Reset    3    5
+    ${key_menu}=    Enter Key Management And Return Construction
+    Enroll DB Signature    ${key_menu}    RSA3072    cert.der
+    Save Changes And Reset
 
-    Enter UEFI Shell
-    ${out}=    Execute File In UEFI Shell    signed-hello.efi
-    Should Contain    ${out}    Hello, world!
+    Boot Efi File    signed-hello.efi    RSA3072    Hello, world!
 
 SBO010.003 Check support for rsa4k signed certificates
     [Documentation]    PEM generated with `openssl req -new -x509 -newkey rsa:4096 -subj "/CN=DB-RSA4096/" -keyout DB-RSA4096.key -out DB-RSA4096.pem -days 3650 -nodes -sha256`
     ...    converted to DER using `openssl base64 -d -in DB-RSA4096.cer -out DB-RSA4096.der`
-    Mount ISO As USB In QEMU    ${CURDIR}/../scripts/secure-boot/images/RSA4096.img
+    Skip If    not ${SECURE_BOOT_SUPPORT}    SBO010.003 not supported
+    Skip If    not ${TESTS_IN_FIRMWARE_SUPPORT}    SBO010.003 not supported
+    Mount ISO As USB    ${CURDIR}/../scripts/secure-boot/images/RSA4096.img
     Power On
 
     ${sb_menu}=    Enter Secure Boot Menu And Return Construction
     Enable Secure Boot    ${sb_menu}
-    ${sb_menu}=    Reenter Menu And Return Construction
-    ${advanced_menu}=    Enter Advanced Secure Boot Keys Management And Return Construction    ${sb_menu}
-    Enter Enroll DB Signature Using File In DB Options    ${advanced_menu}
-    Enter Volume In File Explorer    RSA4096
-    Select File In File Explorer    cert.der
-    Save Changes And Reset    3    5
+    ${key_menu}=    Enter Key Management And Return Construction
+    Enroll DB Signature    ${key_menu}    RSA4096    cert.der
+    Save Changes And Reset
 
-    Enter UEFI Shell
-    ${out}=    Execute File In UEFI Shell    signed-hello.efi
-    Should Contain    ${out}    Hello, world!
+    Boot Efi File    signed-hello.efi    RSA4096    Hello, world!
 
 SBO010.004 Check support for ecdsa256 signed certificates
```

Comment:
@m-iwanicki While running SB tests on commit 236b638a06073132378566a98d62c37195c7c83e faced following errors:

```bash
(...)
SBO010.004 Check support for ecdsa256 signed certificates :: PEM g... | FAIL |
'
Command Error Status: Access Denied
FS0:\>' does not contain 'hello, world!'
------------------------------------------------------------------------------
SBO010.005 Check support for ecdsa384 signed certificates :: PEM g... | FAIL |
'
Command Error Status: Access Denied
FS0:\>' does not contain 'hello, world!'
------------------------------------------------------------------------------
SBO010.006 Check support for ecdsa521 signed certificates :: PEM g... | FAIL |
'
Command Error Status: Access Denied
FS0:\>' does not contain 'hello, world!'
------------------------------------------------------------------------------
SBO011.001 Attempt to enroll expired certificate and boot signed i... | FAIL |
No match found for 'exit' in 30 seconds. Output:
------------------------------------------------------------------------------
(...)
```

Reply:
@m-iwanicki I suppose it is because of unsupported ECDSA format?  Or maybe the way ECDSA certificate is being generated is wrong? Faced following issue while verifying manual steps for `SB010.004`, `SB010.005` and `SB010.006`:
![image](https://github.com/Dasharo/open-source-firmware-validation/assets/109155473/4e0b3970-fa9a-4e6c-a979-bede5e1b2615)

Reply:
@DaniilKl ecdsa certs are prepared correctly but we are not able to use them in BIOS, so it looks like this algorithm is just not supported.

Also booting files signed with key that has outdated certificates is working, which is not something that we would expect, hence the test with expired certificate fails also